xstate-chatbot/nodejs/src/channel/gupshup.js

```diff
@@ -0,0 +1,65 @@
+const fetch = require("node-fetch");
+require('url-search-params-polyfill');
+
+class GupShupWhatsAppProvider {
+
+    processMessageFromUser(req) {
+        let reformattedMessage = {}
+        let requestBody = req.body;
+        
+        let type = requestBody.payload.type;
+        let input;
+        if(type === "location") {
+            let location = requestBody.payload.payload;
+            input = '(' + location.latitude + ',' + location.longitude + ')';
+        } else {
+            input = requestBody.payload.payload.text;
+        }
+
+        reformattedMessage.message = {
+            input: input,
+            type: type
+        }
+        reformattedMessage.user = {
+            mobileNumber: requestBody.payload.sender.phone.slice(2)
+        };
+
+        return reformattedMessage;
+    }
+
+    sendMessageToUser(user, outputMessages) {
+        if(!Array.isArray(outputMessages)) {
+            let message = outputMessages;
+            outputMessages = [ message ];
+            console.warn('Output array had to be constructed. Remove the use of deeprecated function from the code. \ndialog.sendMessage() function should be used to send any message instead of any previously used methods.');
+        }
+        for(let message of outputMessages) {
+            let phone = user.mobileNumber;
+
+            let url = "https://api.gupshup.io/sm/api/v1/msg";
```

Comment:
Regarding Gupshup Adapter, I need your advice. I created the Gupshup Adapter only because they are offering a free account and for the initial demos it was easier rather than requesting ValueFirst Partner. We have not partnered with GupShup. In the current state, the Gupshup adapter is not fully featured compared to other adapters. Now, I don't want to invest any more time improving it and adding all features. As it is half-baked, what do you suggest
1. Keeping it as is
2. Removing it from the project

Reply:
Keep it as template, but add these comments to the file at least

Reply:
Done
https://github.com/egovernments/core-services/pull/591/commits/01936e32097fa2c431fbc76a02f34ffc49c9ee1a#diff-8a317f231c13b0b0d6b33f0b5031f94fff96a2601332462836931b0c2ebf9d06